ui/js/actions/app.js

```diff
@@ -129,22 +130,56 @@ export function doCancelUpgrade() {
 export function doCheckUpgradeAvailable() {
   return function(dispatch, getState) {
     const state = getState();
+    dispatch({
+      type: types.CHECK_UPGRADE_STARTED,
+    });
 
-    lbry.getAppVersionInfo().then(({ remoteVersion, upgradeAvailable }) => {
-      if (upgradeAvailable) {
-        dispatch({
-          type: types.UPDATE_VERSION,
-          data: {
-            version: remoteVersion,
-          },
-        });
+    const success = ({ remoteVersion, upgradeAvailable }) => {
+      dispatch({
+        type: types.CHECK_UPGRADE_COMPLETED,
+        data: {
+          upgradeAvailable,
+          remoteVersion,
+        },
+      });
+
+      // If there's an available upgrade and the user hasn't skipped it or if there's a newer one, show un upgrade modal
+      if (
+        upgradeAvailable &&
+        (!selectIsUpgradeSkipped(state) ||
+          remoteVersion !== selectRemoteVersion(state))
+      ) {
```

Comment:
Since this function is now firing repeatedly rather than just on first-run, it is worth some consideration as to whether this can fire at an inappropriate time.

For example: what if a modal is already open? Should this overwrite it?

Reply:
Good point! We shouldn't restart the upgrade process if one is already being run.

Reply:
Not just the upgrade process - what if a modal was open displaying an important error message or point of feedback? Only one modal can be open at a time - we don't have a modal stack.

Reply:
Yep! I thought about that when I fixed it.